Docs: Update doc for PostgreSQL authentication (#31434)
grafana/grafana: docs/sources/datasources/postgres.md

### Data source options

Name               | Description
-----------------  | -------------
`Name`             | The data source name. This is how you refer to the data source in panels and queries.
`Default`          | Default data source means that it will be pre-selected for new panels.

> `Host`             | The IP address/hostname and optional port of your PostgreSQL instance. _Do not_ include the database name. The connection string for connecting to Postgres will not be correct and will cause errors.

Reviewer: ```suggestion
`Host`             | The IP address/hostname and optional port of your PostgreSQL instance. _Do not_ include the database name. The connection string for connecting to Postgres will not be correct and it may cause errors.
```

`Database`         | Name of your PostgreSQL database.
`User`             | Database user's login/username
`Password`         | Database user's password

> `SSL Mode`         | This option determines whether or with what priority a secure SSL TCP/IP connection will be negotiated with the server. When SSL Mode is disable, SSL Method and Auth Details would not be visible.

Reviewer: ```suggestion
`SSL Mode`         | Determines whether or with what priority a secure SSL TCP/IP connection will be negotiated with the server. When SSL Mode is disabled, SSL Method and Auth Details would not be visible.
```

> `SSL Method`       | Determines whether the SSL Auth details will be configured as a file path or file content.

Reviewer 2: Maybe this field combination is better as `SSL Auth Details Method`  and `SSL Auth Details Value` as both these configuration options are tied together.

Probably also good to mention `(Grafana v7.5+).` at the end too.

> `SSL Method`       | This option determines SSL Auth details would be configured as file path or file content

Reviewer: ```suggestion
`SSL Method`       | Determines whether the SSL Auth details will be configured as a file path or file content.
```

`SSL Auth Details` | File path or file content of SSL root certificate, client certificate and client key
`Max open`         | The maximum number of open connections to the database, default `unlimited` (Grafana v5.4+).
`Max idle`         | The maximum number of connections in the idle connection pool, default `2` (Grafana v5.4+).
`Max lifetime`     | The maximum amount of time in seconds a connection may be reused, default `14400`/4 hours (Grafana v5.4+).

> `Version`          | This option determines which functions are available in the query builder (only available in Grafana 5.3+).

Reviewer: ```suggestion
`Version`          |Determines which functions are available in the query builder (only available in Grafana 5.3+).
```

> `TimescaleDB`      |A time-series database built as a PostgreSQL extension. If enabled, Grafana uses `time_bucket` in the `$__timeGroup` macro to display TimescaleDB specific aggregate functions in the query builder (only available in Grafana 5.3+).

Reviewer: ```suggestion
`TimescaleDB`      |A time-series database built as a PostgreSQL extension. When enabled, Grafana uses `time_bucket` in the `$__timeGroup` macro to display TimescaleDB specific aggregate functions in the query builder (only available in Grafana 5.3+).
```

> `TimescaleDB`      | TimescaleDB is a time-series database built as a PostgreSQL extension. If enabled, Grafana will use `time_bucket` in the `$__timeGroup` macro and display TimescaleDB specific aggregate functions in the query builder (only available in Grafana 5.3+).

Reviewer: ```suggestion
`TimescaleDB`      |A time-series database built as a PostgreSQL extension. If enabled, Grafana uses `time_bucket` in the `$__timeGroup` macro to display TimescaleDB specific aggregate functions in the query builder (only available in Grafana 5.3+).
```
